File: tests/csharp/linq_lambdas/linq_aggregate.cs
// vybe-test: csharp/linq_lambdas/linq_aggregate
// origin: languages/csharp/tests/csharp/test_linq_lambdas.rs

using static __Harness;

var nums = new List<int> { 1, 2, 3, 4, 5 }
;
var product = nums.Aggregate(1, (acc, x) => acc * x);
__P((product).ToString());
__Check("120");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/vb/vb_linq_aggregate/linq_aggregate.vb
' vybe-test: vb/vb_linq_aggregate/linq_aggregate
' origin: languages/vb/tests/vb/test_vb_linq_aggregate.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Imports System.Linq

Module M
    Sub Main()
        Dim numbers = {1, 2, 3, 4}
        
        Dim query = Aggregate n In numbers Into Sum(), Max(), Min()
        
        __Check(CStr(query.Sum), "10")
        __Check(CStr(query.Max), "4")
        __Check(CStr(query.Min), "1")
    End Sub
End Module
